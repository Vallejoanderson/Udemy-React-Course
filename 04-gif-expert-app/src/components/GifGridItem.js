import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {

    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={ url } alt={ title }/>
            <p> { title }</p>
        </div>
    )
}

GifGridItem.propTypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
 }

/*  Tarea
    1. Enzyme
    2. Enzyme to Json
    3. Debe de mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSsnapshot()
*/