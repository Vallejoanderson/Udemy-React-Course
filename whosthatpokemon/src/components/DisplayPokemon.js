

export const DisplayPokemon = async({ pokemones }) => {
    // const randomNumber = Math.floor(Math.random() * 3);
    // const image = await pokemones[randomNumber].image;
    // console.log(pokemones);
    return (
        <img src="" alt="Random pokemon" />
    );
}