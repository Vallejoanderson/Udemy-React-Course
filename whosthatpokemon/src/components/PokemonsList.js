

export const PokemonsList = ( { pokemons } ) => {
    const checkChoice = () => {
        
    }

    return(
        <ol>
            {
                pokemons.map( pokemon => 
                    <button onClick={ () => { console.log( pokemon.name ) } }>{ pokemon.name }</button>)
            }
        </ol>
    );
}