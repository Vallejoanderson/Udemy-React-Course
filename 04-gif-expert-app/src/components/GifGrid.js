import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ( { category } ) => {

    const [images, setImages] = useState([])
    
    useEffect( () => {
        getGifs( category )
            .then( imgs => setImages( imgs) ) 
    }, [ category ] )

    
    //getGifs();

    return (
        <>
            <h3>{  category }</h3>
            <div className="card-grid">
                {
                        images.map( img => //Desestructuraci√≥n de objecto images
                            <GifGridItem 
                                key={img.id}
                                { ...img }
                            />
                        )
                }
            </div>
        </>
    )
}
