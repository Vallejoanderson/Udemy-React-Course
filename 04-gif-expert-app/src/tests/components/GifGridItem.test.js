import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

import '@testing-library/jest-dom';

describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'Un titulo';
    const url = 'https://localhost/image.jpg';
    const wrapper = shallow( 
        <GifGridItem 
            title = { title }
            url = { url }
        /> 
    );
    
    test('Debe de mostrar el componente <GifGridItem />', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });


    test('Debe de tener este parrafo con el titulo', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
        
    })
    

    test('Debe de tener la image igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    })
    

    test('Debe de tener la clase animate__zoomIn', () => {
        
        const div = wrapper.find('div');

        console.log( div.props().className );

        expect( div.props().className ).toContain('animate__zoomIn');
    })

})