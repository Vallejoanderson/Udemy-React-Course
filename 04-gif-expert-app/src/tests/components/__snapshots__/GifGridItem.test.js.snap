// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/image.jpg"
  />
  <p>
     
    Un titulo
  </p>
</div>
`;
